### ISSUE TITLE
Threema Web on iPads

### ISSUE THREAD

**Issue by dbrgn** [threema-ch/threema-web#512]
According to this tweet:

https://twitter.com/krolli53/status/1005083421092134912

...the layout is broken on an iPad:

![image](https://user-images.githubusercontent.com/105168/41173380-f3b88b62-6b56-11e8-8bc5-ffccc325a7f1.png)

**Comment by ovalseven8** [comment]
Is not this something that should be fixed before the release of Threema Web for iOS?

**Comment by dbrgn** [comment]
The layout problem is that - when showing the keyboard - the `vh` unit (viewport height) is not adjusted by mobile Safari. I'd consider that a bug in Safari.

There's also a lot of other buggy behavior in the mobile Safari. For example, after sending some text, the buttons around the compose area stop working. It seems that somehow the event handlers break.

Maybe it's this issue: https://stackoverflow.com/questions/34575510/angular-ng-click-issues-on-safari-with-ios-8-3 Feels like adding IE6 fixes... :roll_eyes:

**Comment by dbrgn** [comment]
Tested again on an iPad Pro 2016 with iOS 12. With that hardware Threema Web runs quite smoothly. (The first iPad we tested with was an iPad Mini 3 from 2014, apparently that hardware was simply not sufficient.)

Copying text to clipboard didn't work, this is fixed in #626.

Unfortunately `vh` units are broken on mobile Safari though, as already mentioned earlier: https://bugs.webkit.org/show_bug.cgi?id=141832 Furthermore the keyboard is overlaying the screen. Apparently it's not possible to fix this without horrible and brittle hacks: https://stackoverflow.com/questions/2593139/ipad-web-app-detect-virtual-keyboard-using-javascript-in-safari It can be worked around by the user by manually scrolling down along the sides of the screen.

Therefore I'll close this issue for now. If someone manages to fix the layout on Safari without resorting to querying JS size change events and without breaking something else in the layout, then pull requests are welcome! Other bugs in Threema Web on mobile Safari should be reported as separate issues.
